plugin/pkg/scheduler/extender.go

```diff
@@ -82,53 +83,89 @@ func NewHTTPExtender(config *schedulerapi.ExtenderConfig, apiVersion string) (al
 		Timeout:   config.HTTPTimeout,
 	}
 	return &HTTPExtender{
-		extenderURL:    config.URLPrefix,
-		apiVersion:     apiVersion,
-		filterVerb:     config.FilterVerb,
-		prioritizeVerb: config.PrioritizeVerb,
-		weight:         config.Weight,
-		client:         client,
+		extenderURL:      config.URLPrefix,
+		filterVerb:       config.FilterVerb,
+		prioritizeVerb:   config.PrioritizeVerb,
+		weight:           config.Weight,
+		client:           client,
+		nodeCacheCapable: config.NodeCacheCapable,
 	}, nil
 }
 
 // Filter based on extender implemented predicate functions. The filtered list is
 // expected to be a subset of the supplied list. failedNodesMap optionally contains
 // the list of failed nodes and failure reasons.
-func (h *HTTPExtender) Filter(pod *v1.Pod, nodes []*v1.Node) ([]*v1.Node, schedulerapi.FailedNodesMap, error) {
-	var result schedulerapi.ExtenderFilterResult
+func (h *HTTPExtender) Filter(pod *v1.Pod, nodes []*v1.Node, nodeNameToInfo map[string]*schedulercache.NodeInfo) ([]*v1.Node, schedulerapi.FailedNodesMap, schedulerapi.Annotations, error) {
+	var (
+		result         schedulerapi.ExtenderFilterResult
+		nodeList       *v1.NodeList
+		nodeNames      *[]string
+		nodeResult     []*v1.Node
+		podAnnotations schedulerapi.Annotations
+		args           *schedulerapi.ExtenderArgs
+	)
 
 	if h.filterVerb == "" {
-		return nodes, schedulerapi.FailedNodesMap{}, nil
+		return nodes, schedulerapi.FailedNodesMap{}, nil, nil
 	}
 
-	nodeItems := make([]v1.Node, 0, len(nodes))
-	for _, node := range nodes {
-		nodeItems = append(nodeItems, *node)
+	if h.nodeCacheCapable {
+		nodeNameSlice := make([]string, 0, len(nodes))
+		for _, node := range nodes {
+			nodeNameSlice = append(nodeNameSlice, node.Name)
+		}
+		nodeNames = &nodeNameSlice
```

Comment:
@timothysc @ravigadde synced up with Ravi offline ... backed out pod annotations related changes and pushed the updated changes. Thanks for the review.